wandb/sdk/wandb_run.py

```diff
@@ -502,38 +525,22 @@ def _init(
         if self._settings._require_service:
             self._attach_id = self._settings.run_id
 
-    def _populate_sweep_or_launch_config(
-        self, sweep_config: Optional[Dict[str, Any]]
-    ) -> None:
-        self._launch_artifact_mapping: Dict[str, Any] = {}
-        self._unique_launch_artifact_sequence_names: Dict[str, Any] = {}
-        if sweep_config:
-            self._config.update_locked(
-                sweep_config, user="sweep", _allow_val_change=True
-            )
+    def _set_iface_pid(self, iface_pid: int) -> None:
+        self._iface_pid = iface_pid
 
-        if self._settings.launch and (
-            os.environ.get("WANDB_CONFIG") is not None
-            or os.environ.get("WANDB_ARTIFACTS") is not None
-        ):
-            if os.environ.get("WANDB_CONFIG") is not None:
-                try:
-                    new_config = json.loads(os.environ.get("WANDB_CONFIG", "{}"))
-                    self._config.update_locked(
-                        new_config, user="launch", _allow_val_change=True
-                    )
-                except (ValueError, SyntaxError):
-                    wandb.termwarn("Malformed WANDB_CONFIG, using original config")
-            if os.environ.get("WANDB_ARTIFACTS") is not None:
-                try:
-                    artifacts: Dict[str, Any] = json.loads(
-                        os.environ.get("WANDB_ARTIFACTS", "{}")
-                    )
-                    self._initialize_launch_artifact_maps(artifacts)
-                except (ValueError, SyntaxError):
-                    wandb.termwarn(
-                        "Malformed WANDB_ARTIFACTS, using original artifacts"
-                    )
+    def _set_iface_port(self, iface_port: int) -> None:
+        self._iface_port = iface_port
+
+    def _handle_launch_artifact_overrides(self) -> None:
+        if self._settings.launch and (os.environ.get("WANDB_ARTIFACTS") is not None):
```

Comment:
should the `WANDB_ARTIFACTS` be part of the settings?

Reply:
This needs to be an environment variable so we can swap artifacts outside of the config in the launch context without requiring a build step.